packages/mdc-textfield/foundation.js

```diff
@@ -206,15 +209,16 @@ class MDCTextFieldFoundation extends MDCFoundation {
    * Deactives the Text Field's focus state.
    */
   deactivateFocus() {
-    const {FOCUSED, LABEL_FLOAT_ABOVE, LABEL_SHAKE} = MDCTextFieldFoundation.cssClasses;
+    const {FOCUSED} = MDCTextFieldFoundation.cssClasses;
     const input = this.getNativeInput_();
+    const inputIsEmpty = !input.value && !this.isBadInput_();
```

Comment:
Nit: This variable name is a bit misleading since it's more than just if the value is empty. I'm wondering if we should rename this variable (and the parameter for `deactivateFocus` where it's defined/documented) to something like `shouldPreserveFloat` or something (though that name in particular implies the opposite of its current name).

Reply:
how about `inputIsEmptyAndValid`?